core/src/main/java/tech/tablesaw/interpolation/Interpolator.java

```diff
@@ -0,0 +1,48 @@
+package tech.tablesaw.interpolation;
+
+import tech.tablesaw.columns.Column;
+
+/**
+ * Interpolations that are agnostic of column type.
+ */
+public class Interpolator<T> {
+
+  private final Column<T> col;
+
+  public Interpolator(Column<T> column) {
+    this.col = column;
+  }
+
+  /**
+   * Fills missing values with the next non-missing value
+   */
+  public void backfill() {
+    T lastVal = null;
```

Comment:
couldn't this have been written by sharing the logic from frontFill, but with the iteration reversed?

Reply:
In general, I think this implementation might have been better written as methods on AbstractColumn. Implementing it as a Interpolation class might make it more difficult to add advanced interpolation methods (like linear, mean, etc) that assume continuous variables (or at least ordered variables). I don't know how we could add linear interpolation to a string column, for example.

It was hard for me to communicate the other day because I was in an Uber, but I was thinking of an abstract interpolate() method that took a function argument, perhaps with these two implementations provided by default as named methods that use the generic interpolate.  It's still just a thought at this point as I haven't considered all the details. 

What do you think?

Reply:
yeah, the logic is the same except for reversed iteration order, but I'm not sure how we'd share more code?

Reply:
@lwhite1 I added a linear interpolation so you could see what it would look like with this type of implementation. I think the benefit of this way is that methods are nicely grouped together and you don't need to import any other class

Reply:
I still don't see why it should be a separate class. It will be harder to find

Reply:
The hope was that it'd make things easier to find by making it so that you can see all interpolation methods in a single place. Of course you have to figure out that you need to look under `interpolate` first, but then you'd see all the various methods that are available for filling missing cells. If it's on the column directly then the user would have to look through a much larger set of methods to figure out what interpolation methods are available and that could be a bit overwhelming.

I'm fine with putting them directly on the column if that's clearer though.